### PR TITLE
[20.10 backport] update to golang 1.18.5

### PR DESCRIPTION
**Pull request by thaJeztah**
- depends on https://github.com/moby/moby/pull/43975

Backport of:

- https://github.com/moby/moby/pull/43194
- https://github.com/moby/moby/pull/43498
- https://github.com/moby/moby/pull/43581
    - golang.org/x/sys update was already done in https://github.com/moby/moby/pull/43582
- https://github.com/moby/moby/pull/43676
- https://github.com/moby/moby/pull/43802
- https://github.com/moby/moby/pull/43903

### CODE REVIEW

File: vendor/archive/tar/common.go [hunk 1]
```diff
@@ -538,7 +538,7 @@ type headerFileInfo struct {
 func (fi headerFileInfo) Size() int64        { return fi.h.Size }
 func (fi headerFileInfo) IsDir() bool        { return fi.Mode().IsDir() }
 func (fi headerFileInfo) ModTime() time.Time { return fi.h.ModTime }
-func (fi headerFileInfo) Sys() interface{}   { return fi.h }
+func (fi headerFileInfo) Sys() any           { return fi.h }
```
Comment (thaJeztah) [inline vendor/archive/tar/common.go:541]: I guess this means we require go1.18. Alternative could be to pin archive/tar to the version of go1.17 (the intent was of course to match the version in go's stdlib

**Review by neersighted — APPROVED** [review]
Looks like you beat me to this PR -- I was just preparing a version locally when I checked the tracker.

This looks good to me -- I don't think the hard dependency on Go 1.18 should really be an issue given 1.17 is EOL, and ideally we'd use the stdlib code from 1.18. At least to me, using the same vendored code is less of a violation of the principle of least surprise.

**Comment by thaJeztah** [comment]
Yes, we managed to get rid of the vendor in master, but not sure it's worth backporting to 20.10

**Comment by thaJeztah** [comment]
Windows failure in CI is unrelated; not sure if it's worth kicking CI for that one (as it'll take another 1,5 hours);

```
=== FAIL: github.com/docker/docker/integration-cli TestDockerSuite/TestStartReturnCorrectExitCode (9.64s)
[2022-08-18T16:26:43.093Z]     docker_cli_start_test.go:209: assertion failed: expected an error, got nil
[2022-08-18T16:26:43.093Z]     --- FAIL: TestDockerSuite/TestStartReturnCorrectExitCode (9.64s)
```

**Review by corhere — APPROVED** [review]

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**